### DISCUSSION TITLE
Alpine based page doesn't pass W3C validator markup validation service

### DISCUSSION THREAD

**Discussion by driss-chelouati** [alpinejs/alpine#2042] [7. Bugs (archived)] [answered]
Iam building an application template with Alpine v3. Due to some project constraints, Iam forced to test out W3C validation on each application page. It fails with a lot of warning and errors because of Alpine custom attributes (x-*) and bindings like :class.
Is there something to do about this? see attached screenshot
![attributes](https://user-images.githubusercontent.com/86636408/132776744-0653b1ac-d2a4-4308-9c48-755130194a38.png)

**Comment by SimoTod** [comment, marked as answer]
Alpine v3 supports custom prefixes.
```html
<script>
document.addEventListener('alpine:init', () => Alpine prefix('data-x-'))
</script>
```
`data-*` attributes are allowed and they'll pass validation

**Reply by khanakia** [reply]
I am still getting error for any attribute which contains the `colon :`

w3 validator still giving error 
data-* attribute names must be XML 1.0 4th ed. plus Namespaces NCNames

<img width="1023" alt="Screenshot 2022-09-26 at 3 52 44 PM" src="https://user-images.githubusercontent.com/2174170/192253682-3dd4331d-d2a2-457c-9c25-4b70d055a221.png">


You aren't allowed colons (:) in the names of data-* attributes.

**Reply by SimoTod** [reply]
Alpine is not w3c compliant out of the box. You can use x-bind and define the attributes in the js side but it's probably not as immediate as it would be in the html

**Reply by ekwoka** [reply]
> Alpine is not w3c compliant out of the box. You can use x-bind and define the attributes in the js side but it's probably not as immediate as it would be in the html

You do

```
x-bind="{':style': 'display: none;'}"
```

**Reply by JorickPepin** [reply]
> > Alpine is not w3c compliant out of the box. You can use x-bind and define the attributes in the js side but it's probably not as immediate as it would be in the html
> 
> You do
> 
> ```
> x-bind="{':style': 'display: none;'}"
> ```

Do you know if it's possible to do the same with `x-on` ? Something like
```
x-on="{':click': alert()}"
```
instead of
```
x-on:click="alert()"
```

I can't get it to work.

**Reply by ekwoka** [reply]
x-bind="{'@click'() { do stuff } }"

**Reply by Zmove** [reply]
Would be great to have an alpine 4 W3C compatible. There is probably a way to find a syntax that can fit w3c standard without losing features
